src/eigen_gmres.cc

```diff
@@ -30,7 +30,8 @@ bool pipenetwork::EigenGMRES::solve() {
   gmres.setTolerance(tolerance_);
 
   x_diff = gmres.solve(*vec_b_);
-  x_diff = x_diff.array() * vrestraints_.array();
+  x_diff = x_diff.array();
+  //    x_diff = x_diff.array() * vrestraints_.array();
```

Comment:
So we no longer need to apply any restraints?

Reply:
Yes! No need for constraints since it is included inside the jacobian.